pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -458,13 +446,13 @@ func (c *Controller) runDeleteShootFlow(o *operation.Operation, errorContext *er
 		)
 
 		destroyInternalDomainDNSRecord = g.Add(flow.Task{
-			Name:         "Destroying internal domain DNS record",
-			Fn:           flow.TaskFn(botanist.DestroyInternalDomainDNSRecord).DoIf(dnsEnabled),
-			Dependencies: flow.NewTaskIDs(syncPoint),
+			Name:         "Destroying internal DNS entry",
+			Fn:           flow.TaskFn(component.OpWaiter(botanist.Shoot.Components.DNS.InternalEntry).Destroy),
+			Dependencies: flow.NewTaskIDs(syncPoint, destroyNginxIngressDNSRecord),
```

Comment:
```suggestion
			Dependencies: flow.NewTaskIDs(syncPoint),
```

`destroyNginxIngressDNSRecord` is part of `syncPoint`